src/crimson/os/seastore/seastore.cc

```diff
@@ -194,15 +194,49 @@ SeaStore::read_errorator::future<ceph::bufferlist> SeaStore::readv(
   return read_errorator::make_ready_future<ceph::bufferlist>();
 }
 
+using crimson::os::seastore::omap_manager::BtreeOMapManager;
+
 SeaStore::get_attr_errorator::future<ceph::bufferlist> SeaStore::get_attr(
   CollectionRef ch,
   const ghobject_t& oid,
   std::string_view name) const
 {
   auto c = static_cast<SeastoreCollection*>(ch.get());
   logger().debug("{} {} {}",
-                __func__, c->get_cid(), oid);
-  return crimson::ct_error::enoent::make();
+		 __func__, c->get_cid(), oid);
+  using get_attr_ertr = TransactionManager::base_ertr::extend<
+    crimson::ct_error::enoent>;
+  return repeat_with_onode<ceph::bufferlist>(
+    c, oid, [=](auto &t, auto& onode)
+    -> get_attr_ertr::future<ceph::bufferlist> {
+    auto& layout = onode.get_layout();
+    if (name == OI_ATTR && layout.oi_size) {
+      ceph::bufferlist bl;
+      bl.append(ceph::bufferptr(&layout.oi[0], layout.oi_size));
+      return seastar::make_ready_future<ceph::bufferlist>(std::move(bl));
+    }
+    if (name == SS_ATTR && layout.ss_size) {
+      ceph::bufferlist bl;
+      bl.append(ceph::bufferptr(&layout.ss[0], layout.ss_size));
+      return seastar::make_ready_future<ceph::bufferlist>(std::move(bl));
+    }
+    return seastar::do_with(
+      BtreeOMapManager(*transaction_manager),
+      layout.xattr_root.get(),
+      std::string(name),
+      [&t, this](auto &manager, auto& root, auto& name) {
+      return manager.omap_get_value(root, t, name).safe_then([](auto opt)
```

Comment:
Root might be null if we haven't actually stored anything other than OI_ATTR and SS_ATTR (and they're small enough), right?